arrow-cast/src/cast/mod.rs

```diff
@@ -1229,6 +1230,13 @@ pub fn cast_with_options(
                 cast_byte_container::<BinaryType, LargeBinaryType>(&binary)
             }
             Utf8View => Ok(Arc::new(StringViewArray::from(array.as_string::<i32>()))),
+            BinaryView => Ok(Arc::new(BinaryViewArray::from(
+                array
+                    .as_string::<i32>()
+                    .into_iter()
+                    .map(|x| x.map(|x| x.as_bytes()))
+                    .collect::<Vec<_>>(),
+            ))),
```

Comment:
I think this code does a string-by-string copy 

I played around and I think the copy can be avoided by converting to StringViewArray which can reuse the buffer and then dropping the knowledge about utf8

```suggestion
            BinaryView => Ok(Arc::new(StringViewArray::from(array.as_string::<i32>()).to_binary_view())),
```